### DISCUSSION TITLE
How to use loaders/transform without file extension

### DISCUSSION THREAD

**Discussion by elfjes** [visgl/deck.gl#7213] [Q&A]
I have a prop that needs data from a url. This url does not include a file extension. I'd now like to make this prop an `async` prop so that it uses deck's `fetch` functionality. After downloading I need to transform this data. I see I have two options:

 * use a Loader. However, I'm don't seem to be able to trigger a loader when the url does not have a file extension.
 * use defaultProps.prop.transform. However, if there is not loader matched, deck.gl will error `No valid loader found`

How can I download and transform data coming from an arbitrary url?

**Comment by ibgreen** [comment]
Are you able to supply the `mimeType` or `fallbackMimeType` loader options to the layer loadOptions?

https://loaders.gl/modules/core/docs/api-reference/loader-options

**Reply by elfjes** [reply]
Thanks, I'll have a look. I'm a little bit uncertain what value I should give to `mimeType` though. The docs say `'application/x-<loader.id>'` but `loader.id` is not part of a loader (iirc). Should it be `loader.name`?

**Reply by elfjes** [reply]
ah, right. I need to supply a `mimeTypes` field to my loader, and then it automatically selects my loader based on mimetype. Thanks for pointing me in the right direction 😄
